### PR TITLE
Session upload and download methods on session objects.

### PR DESCRIPTION
**Pull request by hpohekar**
This  feature is for internal use only.
Added _Uploader class to session.py.
Added upload and download methods to both Session and _BaseSession class.
Added test cases for both Session and _BaseSession class in test_uploader.py.
Tested code in Ansys Lab and working fine.

### CODE REVIEW

File: tests/test_uploader.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from ansys.fluent.core import launch_fluent
+
+
+def test_base_session():
+    base_session = launch_fluent(cleanup_on_exit=False, mode="meshing")
+    with pytest.raises(Exception) as e_info:
+        base_session._upload(
+            "D:/PythonPractise/RealWithUnits/ConversionTable.py", "testNow.py"
```
Comment (mkundu1) [inline tests/test_uploader.py:10]: Is the path `D:/..` from local machine? This won't accesible from linux docker container. Plase use a path relative to `pyfluent.EXAMPLES_PATH`

File: src/ansys/fluent/core/session.py [hunk 2]
```diff
@@ -348,3 +375,63 @@ def root(self):
...
+        Download a file from the server.
+    """
+
+    def __init__(self, pim_instance):
+        self.pim_instance = pim_instance
+
+        try:
+            upload_server = self.pim_instance.services["http-simple-upload-server"]
+        except AttributeError:
+            print("\nPIM is not installed or not authorized.\n")
```
Comment (mkundu1) [inline src/ansys/fluent/core/session.py:414]: `LOG.error` would be preferred.

File: tests/test_uploader.py (hunk 1 above)
Comment (hpohekar) [reply tests/test_uploader.py:10]: Ok. Thank you.

File: src/ansys/fluent/core/session.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py:414]: Ok. Thank you.

Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Done.

File: tests/test_uploader.py (hunk 1 above)
Comment (hpohekar) [reply tests/test_uploader.py]: Done.

**Review by mkundu1 — APPROVED** [review]

File: tests/test_uploader.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        base_session._upload(
+            pyfluent.EXAMPLES_PATH + "/mixing_elbow.py", "test_upload_download.py"
+        )
+        base_session._download("test_upload_download.py", pyfluent.EXAMPLES_PATH)
+    base_session.exit()
+
+
+def test_session(new_mesh_session):
+    session = new_mesh_session
+    with pytest.raises(Exception) as e_info:
```
Comment (seanpearsonuk) [inline tests/test_uploader.py:19]: if asserting an exception will be raised, assert it for each method that's expected to raise.

File: src/ansys/fluent/core/session.py [hunk 4]
```diff
@@ -348,3 +376,61 @@ def root(self):
...
+        expanded_file_path = os.path.expandvars(file_path)
+        upload_file_name = remote_file_name or os.path.basename(expanded_file_path)
+        self.file_service.upload_file(expanded_file_path, upload_file_name)
+
+    def download(self, file_name: str, local_file_path: str = None):
+        """Downloads a file from the server."""
+        if self.file_service.file_exist(file_name):
+            self.file_service.download_file(file_name, local_file_path)
+        else:
+            raise FileNotFoundError("Uploaded remote file name is different.")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:436]: Can we just say that the remote file does not exist?

File: src/ansys/fluent/core/session.py [hunk 5]
```diff
@@ -319,6 +335,18 @@ def __dir__(self):
             )
         )
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:337]: are `_upload` and `_download` simply non-throwing no-op methods if pim is unconfigured? That is the requirement.

File: tests/test_uploader.py (hunk 3 above)
Comment (hpohekar) [reply tests/test_uploader.py]: Done.

File: src/ansys/fluent/core/session.py (hunk 4 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Yes, we can. Done.

File: src/ansys/fluent/core/session.py (hunk 5 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py:337]: Yes.  
In following usage, we do not have single dependency on PIM or it's instance, it's doesn't matter whether PIM is installed or not. 

Here ```self._remote_instance``` is act as actual ```self._pim_instance```.

```
import ansys.fluent.core as pyfluent
from ansys.fluent.core import launch_fluent

meshing_session = launch_fluent(cleanup_on_exit=False, meshing_mode=True)
pyfluent.session.Session._uploader=None
pyfluent.session.Session._upload=_upload
pyfluent.session.Session._download=_download
meshing_session._upload('upload/exhaust_system11.ipynb', remote_file_name='exh6.ipynb')
meshing_session._download('exh6.ipynb', local_file_path='download')
```

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**